x/nexus/exported/types.go

```diff
@@ -364,3 +366,27 @@ func (m WasmMessage) ValidateBasic() error {
 func (m WasmMessage) GetSigners() []sdk.AccAddress {
 	return []sdk.AccAddress{}
 }
+
+// WasmBytes is a wrapper around []byte that gets JSON marshalized as an array
+// of uint16 instead of base64-encoded string
+type WasmBytes []byte
+
+// MarshalJSON implements json.Marshaler
+func (bz WasmBytes) MarshalJSON() ([]byte, error) {
+	return json.Marshal(slices.Map(bz, func(b byte) uint16 { return uint16(b) }))
+}
+
+// UnmarshalJSON implements json.Unmarshaler
+func (bz *WasmBytes) UnmarshalJSON(data []byte) error {
+	var arr []uint16
+	if err := json.Unmarshal(data, &arr); err != nil {
+		return err
+	}
+
+	*bz = make([]byte, len(arr))
+	for i, v := range arr {
+		(*bz)[i] = byte(v)
```

Comment:
how does uint16 -> uint8 conversion work in case of overflow?

Reply:
`uint8(uint16(256)) == 0`

Reply:
it just takes the first 8 bits.